addon/components/connect.js

```diff
@@ -1,98 +1,113 @@
 import Ember from 'ember';
 
 const {
+  assert,
   computed,
   defineProperty,
+  getProperties,
   inject: { service },
-  isEmpty,
   run
 } = Ember;
 
-export default (stateToComputed, dispatchToActions=() => ({})) => {
+/**
+  Returns a list of keys that have different values between the two objects.
+
+  @method changedKeys
+  @return {Array} keys that have changed
+  @private
+*/
+function changedKeys(state, newState) {
+  return Object.keys(state).filter(key => {
+    return state[key] !== newState[key];
+  });
+}
+
+/**
+  Creates a read-only computed property for accessing redux state.
+
+  @method computedReduxProperty
+  @return {Function} an Ember computed property
+  @private
+*/
+function computedReduxProperty(key, getState) {
+  return computed({
+    get: () => getState()[key],
+    set: () => assert(`Cannot set redux property "${key}".`)
```

Comment:
While I was here, redefining the getter, I figured I might as well prevent `set`s with a more descriptive error per #55.